src/java.base/share/classes/sun/security/pkcs12/PKCS12KeyStore.java

```diff
@@ -1489,18 +1480,30 @@ private void populateAttributes(Entry entry) {
     private byte[] calculateMac(char[] passwd, byte[] data)
         throws IOException
     {
-        byte[] mData;
-        String algName = macAlgorithm.substring(7);
+        final byte[] mData;
+        final AlgorithmParameterSpec params;
+        final String macString;
+        final String algName;
+        final MacData macData;
+
+        if (macAlgorithm.equals("PBMAC1")) {
```

Comment:
The PBMAC1 algorithms are already defined in the standard algorithm names spec, see https://download.java.net/java/early_access/jdk25/docs/specs/security/standard-names.html#mac-algorithms

The default value of the `keystore.pkcs12.macAlgorithm` security property in the `java.security`file should be changed to "PBEWithHmacSHA256" as part of this change.

So you don't need to check if the algorithm is "PBMAC1", just use the algorithm that the property is set to.

Reply:
I thought we will keep the default unchanged this time.